### PR TITLE
Add symbol_visibility keyword argument

### PR DESCRIPTION
**Pull request by jpakkane**
I really don't like the choice duplication in this one, but importing compiler stuff to build is even worse. Suggestions welcome.

### CODE REVIEW

File: mesonbuild/compilers/compilers.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 from .. import mlog
 from .. import mesonlib
 from ..mesonlib import EnvironmentException, MesonException, version_compare, Popen_safe
+from reportlab.graphics.barcode import common
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__

[F401] 'reportlab.graphics.barcode.common' imported but unused

**Comment by nirbheek** [comment]
I wonder if we should prefix it with `gcc_` or something. We should also mention in the docs that it has no effect with Visual Studio.

**Comment by jpakkane** [comment]
It can't really be `gcc_` because Clang does the same thing. And presumably every *nix compiler that supports symbol visibility.

**Comment by nirbheek** [comment]
> It can't really be gcc_ because Clang does the same thing.

Yeah, but we could say that it's 'GCC syntax' and that Clang and ICC follow GCC syntax.

**Comment by nirbheek** [comment]
For instance, Clang supports both GCC syntax for options and VS syntax, but we still prefix our stuff with `vs_` for MSVC-specific things.

**Comment by jpakkane** [comment]
Also, should there be an option `reallyhidden` or something that is `hidden` and adds `-fvisibility-inlines-hidden`?

**Comment by nirbheek** [comment]
> Also, should there be an option reallyhidden or something that is hidden and adds -fvisibility-inlines-hidden?

Bikesheds are my favourite! How about `morehidden`?

**Comment by jpakkane** [comment]
Maybe something like `inlinehidden` since that is the name of the GCC argument.

**Comment by jpakkane** [comment]
This should be ready for review now.

File: docs/markdown/snippets/visibility.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Keyword argument for GNU symbol visibility
+
+Build targets got a new keyword, `symbol_visibility` that controls how
```
Comment (nirbheek) [inline docs/markdown/snippets/visibility.md]: I just noticed this typo, right after you pushed it 😅

**Pull request merged**